src/text/line.rs

```diff
@@ -1148,18 +1148,38 @@ mod tests {
             assert_buffer_eq!(buf, Buffer::with_lines([expected]));
         }
 
+        /// Ensures the rendering also works away from the 0x0 position.
+        #[rstest]
+        #[case::left_6(Alignment::Left, 6, "a🦀bc ")]
+        #[case::center_6(Alignment::Center, 6, "🦀bc🦀")]
+        #[case::right_6(Alignment::Right, 6, " bc🦀d")]
```

Comment:
It's probably a little easier to immediately understand these tests by adding some of the padding around (e.g. "XXXXa🦀bc XXXX") (and the assertion becomes simpler too)

Reply:
I don't think the assertion becomes simpler as the buffer is wider than the truncation width. Truncate to 6 characters and put it somewhere into a 10 width Buffer introduces additional logic.

Creating a Buffer starting at 0x0 and render into 2x2 would likely be simpler, but the diff would be annoying. For this x is only relevant so y could be left at 0.

Side question… why is the Buffer not always based on 0x0 and only takes a size? where is the offset relevant?

Reply:
Simpler because the assertion is just the full contents of the buffer. xx123456xx. This makes it a direct comparison against a constant value rather than a comparison with a mutated expected buffer.

Buffer doesn't start at zero for Fixed viewports which can be anywhere on the screen with the same coordinates as screen coordinates.